### PR TITLE
On file resize, only zero old memory chunks after header has been written

### PR DESCRIPTION
**Pull request by paulo-raca**
There is that very unlike corner case when things crash after the wrapped chunk has been moved to `oldFileLength`, but the header has not been updated yet, causing file corruption.

Zero-ing the old interval after the header has been written fixes this.

### CODE REVIEW

**Comment by f2prateek** [comment]
LGTM. Adding a test case here would be great. There are some checkstyle (https://travis-ci.org/square/tape/builds/181412923) failures but I can fix them separately if you don't get to it.

**Comment by paulo-raca** [comment]
Thanks for the review.
I've fixed the checkstyle warning.

Any suggestions on how to implement the test case?

File: tape/src/main/java/com/squareup/tape2/QueueFile.java [hunk 1]
```diff
@@ -499,6 +496,10 @@ private void expandIfNecessary(long dataLength) throws IOException {
     }
 
     fileLength = newLength;
+
+    if (zero && count != 0) {
```
Comment (f2prateek) [inline tape/src/main/java/com/squareup/tape2/QueueFile.java]: Do we need to check if count is zero? Will it ever be zero?

**Comment by f2prateek** [comment]
Sorry for the delay. This will be tricky to test - likely require some combination of these 2 tests https://github.com/square/tape/blob/master/tape/src/test/java/com/squareup/tape2/QueueFileTest.java#L366-L424.

File: tape/src/main/java/com/squareup/tape2/QueueFile.java (hunk 1 above)
Comment (paulo-raca) [inline tape/src/main/java/com/squareup/tape2/QueueFile.java]: I will be zero when the data is contiguous (not wrapped) before the expansion.

But you are right, we don't need to check it -- ringErase handles it.

**Comment by paulo-raca** [comment]
I implemented the unit test

**Comment by f2prateek** [comment]
LGTM

**Pull request merged**